### PR TITLE
Update oeutil not to pass endorsements when baseline is not null

### PR DESCRIPTION
**Pull request by anche-is-andy**
Signed-off-by: Andy Chen <anche@microsoft.com>

### CODE REVIEW

**Comment by yentsanglee** [comment]
I'd think if baseline and endorsement are both passed it, oe_verify_evidence() should return invalid parameter error?

**Comment by anche-is-andy** [comment]
> I'd think if baseline and endorsement are both passed it, oe_verify_evidence() should return invalid parameter error?

this is a good point, let me add a fix

**Review by RRathna — APPROVED** [review]

**Review by ryanhsu19 — APPROVED** [review]
Looks good to me. Thanks for the work!

File: tools/oeutil/host/generate_evidence.cpp [hunk 1]
```diff
@@ -1212,13 +1212,20 @@ oe_result_t generate_oe_evidence(
...
         {
             oe_policy_t policy = {
                 OE_POLICY_ENDORSEMENTS_BASELINE,
                 (void*)_parameters.baseline,
                 strlen(_parameters.baseline) + 1};
             policies.emplace_back(policy);
+
+            // Endorsements need to be recreated as endorsements baseline is
+            // specified
+            endorsements_buffer = NULL;
```
Comment (qiucwang) [inline tools/oeutil/host/generate_evidence.cpp:1227]: nullptr

**Review by qiucwang — APPROVED** [review]

File: common/attest_plugin.c [hunk 2]
```diff
@@ -326,6 +326,12 @@ oe_result_t oe_verify_evidence(
         }
     }
 
+    // Endorsements buffer conflicts with endorsements baseline policy
+    if (endorsements_buffer && has_endorsements_baseline_policy)
+    {
+        OE_RAISE(OE_INVALID_PARAMETER);
```
Comment (yentsanglee) [inline common/attest_plugin.c:332]: Suggest OE_RAISE_MSG() to give caller more detail in the error

File: tools/oeutil/host/generate_evidence.cpp (hunk 1 above)
Comment (anche-is-andy) [reply tools/oeutil/host/generate_evidence.cpp:1227]: keep NULL to be consistent with other parts, but we should have a cleanup

**Review by yentsanglee — APPROVED** [review]

**Comment by anche-is-andy** [comment]
bors try

**Comment by anche-is-andy** [comment]
bors try+

**Pull request merged**